src/main/java/com/iota/iri/zmq/ZmqTransactionStatsPublisher.java

```diff
@@ -0,0 +1,134 @@
+    private static final String TOTAL_TRANSACTIONS_TOPIC = "t5m2h";
+
+    private static final long MIN_TRANSACTION_AGE_THRESHOLD = Duration.ofMinutes(5).getSeconds();
+    private static final long MAX_TRANSACTION_AGE_THRESHOLD = Duration.ofHours(2).getSeconds();
+
+    private final Logger log = LoggerFactory.getLogger(ZmqTransactionStatsPublisher.class);
+
+    private final Tangle tangle;
+    private final TipsViewModel tipsViewModel;
+    private final TipSelector tipsSelector;
+    private final TimeWindowedApproveeCounter approveeCounter;
+
+    private final MessageQ messageQ;
+
+    private final AtomicBoolean shuttingDown = new AtomicBoolean(false);
+    private Thread thread;
+
+    public ZmqTransactionStatsPublisher(Tangle tangle, TipsViewModel tipsViewModel, TipSelector tipsSelector,
+            MessageQ messageQ) {
+
+        this.tangle = tangle;
+        this.tipsViewModel = tipsViewModel;
+        this.tipsSelector = tipsSelector;
+        this.approveeCounter = new TimeWindowedApproveeCounter(tangle, MIN_TRANSACTION_AGE_THRESHOLD,
+                MAX_TRANSACTION_AGE_THRESHOLD);
+        this.messageQ = messageQ;
+    }
+
+    /**
+     * Starts the publisher.
+     */
+    public void init() {
+        thread = new Thread(getRunnable(), "Transaction Stats Publisher");
+        thread.start();
+    }
+
+    private Runnable getRunnable() {
+        return () -> {
+            while (!shuttingDown.get()) {
+                try {
+                    // the transaction arrival time is in seconds
+                    final long now = Instant.now().getEpochSecond();
+
+                    final long numConfirmed = getConfirmedTransactionsCount(now);
+                    final long numTransactions = getAllTransactionsCount(now);
+
+                    messageQ.publish(CONFIRMED_TRANSACTIONS_TOPIC + " %d", numConfirmed);
+                    messageQ.publish(TOTAL_TRANSACTIONS_TOPIC + " %d", numTransactions);
+                } catch (Exception e) {
+                    log.error("Error while getting transaction counts : {}", e);
+                }
+                try {
+                    Thread.sleep(PUBLISH_INTERVAL);
+                } catch (InterruptedException e) {
+                    log.error("Transaction count interrupted.");
+                }
+            }
+        };
+    }
+
+    private Hash getSuperTip() throws Exception {
+
+        // call the usual tip selection and return the first tip
+        List<Hash> tips = tipsSelector.getTransactionsToApprove(Optional.empty());
+
+        return tips.get(0);
+    }
+
+    private long getConfirmedTransactionsCount(long now) throws Exception {
+
+        return approveeCounter.getCount(now, getSuperTip(), new HashSet<>());
+    }
+
+    private long getAllTransactionsCount(long now) throws Exception {
+
+        // count all transactions in a scalable way, by counting the approvees of all the tips
```

Comment:
This algorithm is O(|tips| * lambda * two_hours). I'm fine with this as a first iteration, but I think the correct way is to index transactions by `arrivalTime` and to filter on that, since it doesn't grow linearly with the number of tips.

Reply:
Eventually, this algorithm should be improved by sorting the transaction by `arrivalTime`. I don't think such a view currently exists. In order to do that, we would need to tap into each arriving tx (maybe via `PersistenceProvider`?) and store them separately in an appropriate data structure. With that in mind, I think the implemented algorithm makes more sense as a first step.